.ci/prepare-settings.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+set -e
+
+mkdir -p .ci-temp
+cp .ci/release-settings.xml .ci-temp/
+
+checkForVariable() {
+  if [[ "$(declare -p "$1")" =~ ^declare\ -x ]]; then
+    echo "Error: Define $1 environment variable"
+    exit 1
+  fi
+
+checkForVariable "SONATYPE_USER"
+checkForVariable "SONATYPE_PWD"
+checkForVariable "SOURCEFORGE_USER"
+checkForVariable "SOURCEFORGE_PWD"
+checkForVariable "GPG_PASSPHRASE"
+checkForVariable "GPG_KEY"
+
+replace() {
+  sed -i "s/$1/$2/g" .ci-temp/release-settings.xml
+}
+
+replace SONATYPE_USER
+replace SONATYPE_PWD
+replace SOURCEFORGE_USER
+replace SOURCEFORGE_PWD
+replace GPG_PASSPHRASE
+replace GPG_KEY
+}
+
+file1="./.ci-temp/release-settings.xml"
+file2="~/.m2/settings.xml"
+
+if cmp -s "$file1" "$file2"; then
```

Comment:
what `cmp` means?

Reply:
`cmp` = comare between two files.
I believe I have to use `diff` command here as both the files are in the different directory?

Reply:
it is fine to have cmp here, as we do not need human readable diff
https://linux.die.net/man/1/cmp
```
✔ ~/java/github/romani/checkstyle [master|✔] 
$ cp .travis.yml .travis.yml.back
✔ ~/java/github/romani/checkstyle [master|…1] 
$ cmp .travis.yml .travis.yml.back 
✔ ~/java/github/romani/checkstyle [master|…1] 
$ vi .travis.yml.back 
✔ ~/java/github/romani/checkstyle [master|…1] 
$ cmp .travis.yml .travis.yml.back 
.travis.yml .travis.yml.back differ: byte 17, line 2
✘-1 ~/java/github/romani/checkstyle [master|…1] 
$
```